### ISSUE TITLE
Linear thumbnail CMYK embedded ICC profile error

### ISSUE THREAD

**Issue by jjonesrs** [libvips/libvips#3564]
## Bug report

**Describe the bug**
Possibly related to #3159, vipsthumbnail with `--linear` fails on the attached image, but succeeds without that option. I've also tried setting the input profile and output profile to cmyk and srgb respectively. It's also possible that the image itself has issues, but it seems to have a profile of Coated GRACoL 2006 (ISO 12647-2:2004) and `vipsImage.Contains("icc-profile-data")` returns true.

The real application uses NetVips, but I believe vipsthumbnail provides simpler reproduction steps.

**To Reproduce**
1. Use the image `sample.tif` in [sample.zip](https://github.com/libvips/libvips/files/12033952/sample.zip)
2. Run vipsthumbnail with the following arguments: `sample.tif --linear`
3. Vips prints `profile incompatible with image`
4. Vips terminates with `icc_import: unable to load or find any compatible input profile`

**Environment**
 - OS: Windows 11 and AWS Lambda Amazon Linux x86_64
 - Vips: 8.14.2

**Comment by kleisauke** [comment]
I was able to reproduce this. It seems that the provided test image is a CMYK image with 5 bands and the embedded ICC profile is not an N-colour profile (i.e. the ICC profile ought to have the `5CLR` signature in this case).
```console
$ vipsheader sample.tif 
sample.tif: 2000x2000 uchar, 5 bands, cmyk, tiffload
$ exiftool -icc_profile:ColorSpaceData sample.tif 
Color Space Data                : CMYK
```

With this patch:
```diff
--- a/libvips/colour/icc_transform.c
+++ b/libvips/colour/icc_transform.c
@@ -580,9 +580,12 @@ vips_image_expected_bands( VipsImage *image )
 			expected_bands = 3;
 			break;
 
+		case VIPS_INTERPRETATION_CMYK:
+			expected_bands = 4;
+			break;
+
 		case VIPS_INTERPRETATION_MULTIBAND:
 		case VIPS_INTERPRETATION_HISTOGRAM:
-		case VIPS_INTERPRETATION_CMYK:
 		case VIPS_INTERPRETATION_MATRIX:
 		case VIPS_INTERPRETATION_FOURIER:
 		default:
```

I'm able to `icc_import` the image with the embedded ICC profile, but this would probally break NCLR support, as noted in: https://github.com/libvips/libvips/pull/3372#discussion_r1125432552.

**Comment by jcupitt** [comment]
Maybe we could add something to find `expected_bands` from the input profile, if there is one? And use the current code if there's no supplied input profile.

**Comment by kleisauke** [comment]
I just opened PR #3565 for this.

**Comment by kleisauke** [comment]
This should be fixed in libvips 8.14.3. Thanks for reporting this!
